SoundSwitch/Framework/Profile/ProfileManager.cs

```diff
@@ -0,0 +1,194 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Windows.Forms;
+using RailSharp;
+using RailSharp.Internal.Result;
+using SoundSwitch.Audio.Manager;
+using SoundSwitch.Audio.Manager.Interop.Enum;
+using SoundSwitch.Common.WinApi;
+using SoundSwitch.Common.WinApi.Keyboard;
+using SoundSwitch.Framework.Configuration;
+using SoundSwitch.Localization;
+
+namespace SoundSwitch.Framework.Profile
+{
+    public class ProfileManager
+    {
+        private readonly Dictionary<HotKeys, ProfileSetting> _profileByHotkey;
+        private readonly Dictionary<string, ProfileSetting> _profileByApplication;
+        private readonly ForegroundProcessChanged _foregroundProcessChanged;
+        private readonly AudioSwitcher _audioSwitcher;
+
+        public IReadOnlyCollection<ProfileSetting> Profiles => AppConfigs.Configuration.ProfileSettings;
+
+        public ProfileManager(ForegroundProcessChanged foregroundProcessChanged, AudioSwitcher audioSwitcher)
+        {
+            _foregroundProcessChanged = foregroundProcessChanged;
+            _audioSwitcher = audioSwitcher;
+            _profileByApplication =
+                AppConfigs.Configuration.ProfileSettings
+                    .Where((setting) => setting.ApplicationPath != null)
+                    .ToDictionary(setting => setting.ApplicationPath.ToLower());
+            _profileByHotkey = AppConfigs.Configuration.ProfileSettings
+                .Where((setting) => setting.HotKeys != null)
+                .ToDictionary(setting => setting.HotKeys);
+        }
+
+        /// <summary>
+        /// Initialize the profile manager. Return the list of Profile that it couldn't register hotkeys for.
+        /// </summary>
+        /// <returns></returns>
+        public Result<ProfileSetting[], VoidSuccess> Init()
+        {
+            RegisterEvents();
+
+            var errors = AppConfigs.Configuration.ProfileSettings
+                .Where(setting => setting.HotKeys != null)
+                .Where(profileSetting => !WindowsAPIAdapter.RegisterHotKey(profileSetting.HotKeys))
+                .ToArray();
+
+            if (errors.Length > 0)
+            {
+                return errors;
+            }
+
+            return Result.Success();
+        }
+
+        private void RegisterEvents()
+        {
+            _foregroundProcessChanged.Event += (sender, @event) =>
+            {
+                _profileByApplication.TryGetValue(@event.ProcessName.ToLower(), out var profile);
```

Comment:
Possible crash see event creation. And how about using `args` as name like everywhere else?